presto-spi/src/main/java/com/facebook/presto/spi/function/FunctionImplementationType.java

```diff
@@ -15,20 +15,28 @@
 
 public enum FunctionImplementationType
 {
-    JAVA(false),
-    SQL(false),
-    THRIFT(true),
-    GRPC(true);
+    JAVA(false, true),
+    SQL(false, true),
+    THRIFT(true, false),
+    GRPC(true, false),
+    CPP(false, false);
```

Comment:
Aren't CPP functions external functions?

Reply:
>  Currently Presto has built-in and non built-in UDFs. For non built-in UDFs, it has three types, SQL, thrift and grpc. For SQL UDF, it will be evaluated in coordinator, and sent to local worker for execution. For thrift and grpc UDFs, they will not be evaluated in coordinator and will be sent to remote servers instead of local workers for evaluation.
In this PR, I add one new UDF type, CPP. This UDF is only used when the local worker is CPP worker instead of JAVA worker. This UDF will not be evaluated in coordinator, and will be sent to local cpp worker for evaluation.

CPP will not be executed using Presto worker, which is contrast to THRIFT and GRPC which use remote UDF server for execution.